README.md

```diff
@@ -7,48 +7,99 @@
 [codecov-url]:https://codecov.io/gh/moxystudio/next-sitemaps-plugin
 [codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/next-sitemaps-plugin/master.svg
 
-This plugin enables your next.js application to have working sitemaps so that search engine crawlers can index all your pages.
+This module provides an API handler and a plugin that enables your next.js application to have working sitemaps so that search engine crawlers can index all your pages.
+
+## Installation
+
+```cmd
+npm install @moxy/next-sitemaps
+```
 
 ## Usage
 
 ### Simple usage
 
+**1. Add the plugin to `next.config.js`**
+
 ```js
+const { withSitemap } = require('@moxy/next-sitemaps');
+
 withPlugins([
     //other plugins..
-    withSitemaps()
+    withSitemap()
     //other plugins...
 ])
 ```
 
-You can also specify options to this plugin:
-
-| Parameter         | Default                | Links                |
-|:-------------|:--------------------------:|--------------------------:|
-| baseUrl           | `/`                   | [More info](#specifying-the-base-url)
-| sitemapsLocation  | `public/sitemaps.xml` | [More info](#specifying-the-target-location)
-| mapDynamicRoutes  | `null`                | [More info](#handling-dynamic-routing)
-
-### Specifying the base url
+**2. Add the API handler on `pages/api/sitemap.js`**
 
 ```js
-withSitemaps({
-    baseUrl: 'https://moxy.studio'
-}),
+import { setupSitemap } from '@moxy/next-sitemaps';
+
+export default setupSitemap({
+    baseUrl: 'https://moxy.studio',
+    mapDynamicRoutes: {
+        '/[id]': () => ['id1', 'id2'],
+    },
+});
 ```
 
-### Specifying the target location
+## API
+
+### withSitemap()
+
+This plugin will match all files and directories inside the next's `/pages` folder and it will map them into urls. Then, a global named `__NEXT_ROUTES__` will be created with all the urls.
+
+> ℹ️ If you are on dev environment and you have added a new route, you must restart your server to update the global `__NEXT_ROUTES__` and hence, get the correct sitemap.
```

Comment:
Your suggestion is a little bit confusing, but I got the idea. 
I think `modify` can confuse developers. You don't need to restart your server when you modify your pages' UI. You just need to restart it if you rename it. 

See my suggestion below and let me know what you think.

> During development, you must restart the server whenever you add, remove or rename an existing page. Otherwise, it won't be reflected in the `/sitemap` response.

Reply:
In fact, this is not true anymore considering this new approach. I'll remove this side note.